### ISSUE TITLE
Layer Normalisation

### ISSUE THREAD

**Issue by CJcool06** [ChrisFuscoMasters/TransformerImpl#3]
Xiong et al. [(On Layer Normalization in the Transformer Architecture)](https://arxiv.org/abs/2002.04745) show that the layer normalisation (LN) position in the vanilla Transformer causes the expected gradients of the parameters near the output layer to be large. Using a large learning rate on these gradients makes training unstable. They propose putting the LN inside of the residual blocks, before the multi-head attention and before the FFN.
